language/documentation/book/src/modules-and-scripts.md

```diff
@@ -71,20 +68,54 @@ module Test {
         Debug::print(&sum)
     }
 }
-}
 ```
 
-The `address 0x42` part specifies that the module will be published under the [account address](./address.md) 0x42 in [global storage](./global-storage-structure.md).
+The `module 0x42::..` part specifies that the module `Test` will be published under the [account address](./address.md) `0x42` in [global storage](./global-storage-structure.md).
 
-Multiple modules can be declared in a single `address` block:
+Modules can also be declared using [named addresses](./address.md). For example:
 
 ```move
-address 0x42 {
-module M { ... }
-module N { ... }
+module TestAddr::Test {
+    struct Example has copy, drop { a: address}
+
+    use Std::Debug;
+    friend TestAddr::AnotherTest;
+
+    public fun print() {
+        let example = Example { a: @TestAddr};
+        Debug::print(&example)
+    }
 }
 ```
 
+Named addresses only exist at the source language level and during compilation
+and named addresses will be fully substituted for their value at the bytecode
+level. Because of this, their value should be treated opaquely and the compiler
+enforces this when possible. For example if we had the following code:
+
+```move=
+script {
+    fun example() {
+        MyAddr::M::foo(@MyAddr);
+    }
+}
+```
+
+and we compiled it with `MyAddr` set to `0xC0FFEE`, then it would be equivalent
+to the following operationally:
+
+```move=
+script {
+    fun example() {
+        0xC0FFEE::M::foo(@0xC0FFEE);
+    }
+}
+```
+
+However at the source level, these _are not equivalent_ -- the function
```

Comment:
Just a general comment, I don't think double hyphens do anything special in Markdown. Instead, you can insert Unicode emdashes and they seem to work fine in my experience (and look much better).